controllers/confidential_handler.go

```diff
@@ -0,0 +1,93 @@
+//
+// Changes the ImageConfigMap, so that the image creation job will create a confidential image.
+// This will happen at the first reconciliation loop, before the image creation job starts.
+//
+// Changes the peer pods configMap to enable confidential.
+// This will happen likely after several reconciliation loops, because it has prerequisites:
+//
+//   - Peer pods must be enabled in the KataConfig.
+//   - The peer pods config map must exist.
+func (r *KataConfigOpenShiftReconciler) enableFeatureConfidential() error {
+
+	// ImageConfigMap
+
+	if err := InitializeImageGenerator(r.Client); err != nil {
+		return err
+	}
+	ig := GetImageGenerator()
+
+	if ig.provider == unsupportedCloudProvider {
+		r.Log.Info("unsupported cloud provider, skipping confidential image configuration")
+	} else {
+		if ig.isImageIDSet() {
+			r.Log.Info("Image ID is already set, skipping confidential image configuration")
+		} else {
+			// Create ImageConfigMap, if it doesn't exist already.
+			if err := ig.createImageConfigMapFromFile(); err != nil {
+				return err
+			}
+
+			// Patch ImageConfigMap.
+			imageConfigMapData := map[string]string{"CONFIDENTIAL_COMPUTE_ENABLED": "yes"}
+			if err := updateConfigMap(r.Client, r.Log, ig.getImageConfigMapName(), OperatorNamespace, imageConfigMapData); err != nil {
+				return err
+			}
+		}
+	}
+
+	// peer pods config
+
+	// Patch peer pods configMap, if it exists.
+	peerpodsCMData := map[string]string{"DISABLECVM": "false"}
+	if err := updateConfigMap(r.Client, r.Log, peerpodsCMName, OperatorNamespace, peerpodsCMData); err != nil {
+		if k8serrors.IsNotFound(err) {
+			// ConfigMap doesn't exist yet, will try again at the next reconcile run.
+		} else {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// disableFeatureConfidential resets config maps to non-confidential values.
+func (r *KataConfigOpenShiftReconciler) disableFeatureConfidential() error {
+
+	if err := InitializeImageGenerator(r.Client); err != nil {
+		return err
+	}
+	ig := GetImageGenerator()
+
+	if ig.provider == unsupportedCloudProvider {
+		r.Log.Info("unsupported cloud provider, skipping confidential image configuration")
+	} else {
+		// Patch ImageConfigMap.
+		imageConfigMapData := map[string]string{"CONFIDENTIAL_COMPUTE_ENABLED": "no"}
+		if err := updateConfigMap(r.Client, r.Log, ig.getImageConfigMapName(), OperatorNamespace, imageConfigMapData); err != nil {
+			if k8serrors.IsNotFound(err) {
+				// Nothing to do, feature is disabled and configMap doesn't exist.
+			} else {
+				return err
+			}
+		}
+	}
+
+	// Patch peer pods configMap, if it exists.
+	peerpodsCMData := map[string]string{"DISABLECVM": "true"}
+	if err := updateConfigMap(r.Client, r.Log, peerpodsCMName, OperatorNamespace, peerpodsCMData); err != nil {
+		if k8serrors.IsNotFound(err) {
+			// Nothing to do, feature is disabled and configMap doesn't exist.
+		} else {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
... and, the comment let aside, this is again the same code as in `enableFeatureConfidential()`.

Either factor out the duplicate code to separate functions or come up with a single `toggleFeatureConfidential()` feature with bool argument.

Reply:
That's how I did it in the beginning, then I split to two functions (enabled/disabled) because I understood the code wasn't clear enough. Happy to know that my first approach was appreciated. I just force-pushed with the refactoring back to a single function.